test/firebase_auth_mocks_test.dart

```diff
@@ -134,6 +146,17 @@ void main() {
     expect(idToken, isNotEmpty);
   });
 
+  test('Returns a hardcoded user token result', () async {
+    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+    final user = auth.currentUser!;
+    final idTokenResult = await user.getIdTokenResult();
+    expect(idTokenResult, isNotNull);
+    expect(
+      idTokenResult.toString(),
```

Comment:
What does `IdTokenResult.toString()` return?

Reply:
There is a custom implementation of `toString()` for `IdTokenResult` ([see here](url)) which returns the following in our test.

```
IdTokenResult(authTime: 2022-10-18 11:55:42.537, claims: {role: admin}, expirationTime: null, issuedAtTime: 2022-10-17 11:55:42.537, signInProvider: phone, token: some_long_token)
```

Normally I wouldn't use `toString` for instance comparison ([equitable](https://pub.dev/packages/equatable) is my usual go-to) but as there's a custom implementation in this case, I thought it might save some future headaches. For example, if the IdTokenResult property names (or our expected values) change in anyway, the test will still work (We don't care about the specific values, only that the two instances match).

That being said, if you feel safer switching to a different method let me know.

Reply:
I like your method. Since they implemented `toString`, using it works well. Would you mind leaving a comment to explain why we're using toString? Something simple like "Using IdTokenResult's implementation of toString" with a link to the code should do.